sdk/schemaregistry/schema-registry-json/src/cache.ts

```diff
@@ -1,75 +1,68 @@
 // Licensed under the MIT license.
 
-import Ajv, { SchemaObject } from "ajv/dist/jtd";
+import Ajv, { SchemaObject } from "ajv";
 import LRUCache from "lru-cache";
 
 import LRUCacheOptions = LRUCache.Options;
 import { logger } from "./logger";
 import { wrapError } from "./errors";
 
-export type AjvSerializer = ReturnType<Ajv["compileSerializer"]>;
-export type AjvDeserializer = ReturnType<Ajv["compileParser"]>;
+export type AjvValidator = ReturnType<Ajv["compile"]>;
 
 export interface CacheEntry {
   /** Schema ID */
   id: string;
-  /** the Ajv serializer */
-  serializer: AjvSerializer;
-  /** the Ajv deserializer */
-  derserializer: AjvDeserializer;
+  /** the Ajv validator */
+  validator: AjvValidator;
 }
 
 const cacheOptions: LRUCacheOptions<string, any> = {
-    max: 128,
-    /**
-     * This is needed in order to specify `sizeCalculation` but we do not intend
-     * to limit the size just yet.
-     */
-    maxSize: Number.MAX_VALUE,
-    sizeCalculation: (_value: any, key: string) => {
-      return key.length;
-    },
-  };
+  max: 128,
+  /**
+   * This is needed in order to specify `sizeCalculation` but we do not intend
+   * to limit the size just yet.
+   */
+  maxSize: Number.MAX_VALUE,
+  sizeCalculation: (_value: any, key: string) => {
+    return key.length;
+  },
+};
 
-  const cacheBySchemaDefinition = new LRUCache<string, CacheEntry>(cacheOptions);
-  const cacheById = new LRUCache<string, AjvDeserializer>(cacheOptions);
+const cacheBySchemaDefinition = new LRUCache<string, CacheEntry>(cacheOptions);
+const cacheById = new LRUCache<string, AjvValidator>(cacheOptions);
 
 const ajv = new Ajv();
 
-export function getCacheById(schemaId: string): AjvDeserializer | undefined {
-    return cacheById.get(schemaId);
+export function getCacheById(schemaId: string): AjvValidator | undefined {
+  return cacheById.get(schemaId);
 }
 
 export function getCacheByDefinition(schema: string): CacheEntry | undefined {
-    return cacheBySchemaDefinition.get(schema);
+  return cacheBySchemaDefinition.get(schema);
 }
 
-function getSchemaSerializers(schema: string , schemaObj?: SchemaObject): { serializer: AjvSerializer, derserializer: AjvDeserializer } {
-    const schemaObject = schemaObj === undefined ? getSchemaObject(schema) : schemaObj;
-    return wrapError(
-      () => {
-        const serializer = ajv.compileSerializer(schemaObject);
-        const derserializer = ajv.compileParser(schemaObject);
-        return { serializer, derserializer };},
-      `Parsing Json schema failed:\n\n\t${schema}\n\nSee 'cause' for more details.`
-    );
-  }
-  
+function getSchemaValidator(schema: string, schemaObj?: SchemaObject): { validator: AjvValidator } {
+  const schemaObject = schemaObj === undefined ? getSchemaObject(schema) : schemaObj;
+  return wrapError(() => {
+    const validator = ajv.compile(schemaObject);
+    return { validator };
+  }, `Validating Json schema failed:\n\n\t${schema}\n\nSee 'cause' for more details.`);
```

Comment:
This step parses the schema to create the validator, there is nothing to validate yet.
```suggestion
  }, `Parsing Json schema failed:\n\n\t${schema}\n\nSee 'cause' for more details.`);
```